src/Microsoft.AspNetCore.ResponseCaching/Interfaces/IResponseCachingCacheKeyModifier.cs

```diff
@@ -5,13 +5,20 @@
 
 namespace Microsoft.AspNetCore.ResponseCaching
 {
-    public interface IResponseCachingCacheKeySuffixProvider
+    public interface IResponseCachingCacheKeyModifier
     {
         /// <summary>
+        /// Create a key segment that is prepended to the default cache key.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/>.</param>
+        /// <returns>The key segment that will be prepended to the default cache key.</returns>
+        string CreatKeyPrefix(HttpContext httpContext);
```

Comment:
I personally care only about prefixing as this is the most obvious visual hint  on a list, and will probably be better supported than suffixes to filter items in the cache. But then why not just let the service change the key freely?

Otherwise, if it's to provide a simple string to append somewhere, an option might be sufficient.